rust/src/treefile.rs

```diff
@@ -206,50 +206,60 @@ fn merge_vec_field<T>(dest: &mut Option<Vec<T>>, src: &mut Option<Vec<T>>) {
     }
 }
 
-/// Given two configs, merge them. Ideally we'd do some macro magic and avoid
-/// listing all of the fields again.
+/// Given two configs, merge them.
 fn treefile_merge(dest: &mut TreeComposeConfig, src: &mut TreeComposeConfig) {
-    merge_basic_field(&mut dest.treeref, &mut src.treeref);
-    merge_basic_field(&mut dest.rojig, &mut src.rojig);
-    merge_vec_field(&mut dest.repos, &mut src.repos);
-    merge_basic_field(&mut dest.selinux, &mut src.selinux);
-    merge_basic_field(&mut dest.gpg_key, &mut src.gpg_key);
-    merge_basic_field(&mut dest.include, &mut src.include);
-    merge_vec_field(&mut dest.packages, &mut src.packages);
-    merge_basic_field(&mut dest.recommends, &mut src.recommends);
-    merge_basic_field(&mut dest.documentation, &mut src.documentation);
-    merge_vec_field(&mut dest.install_langs, &mut src.install_langs);
-    merge_vec_field(&mut dest.initramfs_args, &mut src.initramfs_args);
-    merge_basic_field(&mut dest.boot_location, &mut src.boot_location);
-    merge_basic_field(&mut dest.tmp_is_dir, &mut src.tmp_is_dir);
-    merge_basic_field(&mut dest.default_target, &mut src.default_target);
-    merge_vec_field(&mut dest.units, &mut src.units);
-    merge_basic_field(&mut dest.machineid_compat, &mut src.machineid_compat);
-    merge_basic_field(&mut dest.releasever, &mut src.releasever);
-    merge_basic_field(
-        &mut dest.automatic_version_prefix,
-        &mut src.automatic_version_prefix,
-    );
-    merge_basic_field(&mut dest.mutate_os_release, &mut src.mutate_os_release);
-    merge_basic_field(&mut dest.etc_group_members, &mut src.etc_group_members);
-    merge_basic_field(&mut dest.preserve_passwd, &mut src.preserve_passwd);
-    merge_basic_field(&mut dest.check_passwd, &mut src.check_passwd);
-    merge_basic_field(&mut dest.check_groups, &mut src.check_groups);
-    merge_basic_field(
-        &mut dest.ignore_removed_users,
-        &mut src.ignore_removed_users,
-    );
-    merge_basic_field(
-        &mut dest.ignore_removed_groups,
-        &mut src.ignore_removed_groups,
+    macro_rules! merge_basics {
+        ($field:ident) => {{
+            merge_basic_field(&mut dest.$field, &mut src.$field);
+        }};
+        ($field:ident, $($fields:ident),*) => {{
+             merge_basic_field(&mut dest.$field, &mut src.$field);
+             merge_basics!($($fields),*);
```

Comment:
OK, I see, this look like the macro works off of recursion. Having not used Rust macros much before but looked at https://doc.rust-lang.org/book/second-edition/appendix-04-macros.html, would just this not work instead:

```rust
    macro_rules! merge_basics {
        ( $($field:ident),* ) => {{
            $( merge_basic_field(&mut dest.$field, &mut src.$field); )*
        }};
    };
```

?

It does seem to compile at least!